api/apps/api/src/modules/geo-feature-tags/geo-feature-tags.service.ts

```diff
@@ -0,0 +1,53 @@
+import { InjectRepository } from '@nestjs/typeorm';
+import { GeoFeatureTag } from '@marxan-api/modules/geo-feature-tags/geo-feature-tag.api.entity';
+import { Repository } from 'typeorm';
+import { GeoFeature } from '@marxan-api/modules/geo-features/geo-feature.api.entity';
+import { Either, left, right } from 'fp-ts/lib/Either';
+
+export const featureNotFound = Symbol('feature not found');
+export const featureProjectNotRelated = Symbol(
+  'feature and project not related',
+);
+
+export class GeoFeatureTagsService {
+  constructor(
+    @InjectRepository(GeoFeatureTag)
+    private readonly geoFeatureTagsRepo: Repository<GeoFeatureTag>,
+    @InjectRepository(GeoFeature)
+    private readonly geoFeaturesRepo: Repository<GeoFeature>,
+  ) {}
+
+  async updateTagForFeature(
+    projectId: string,
+    featureId: string,
+    tag: string,
+  ): Promise<
```

Comment:
apologies about not having specified this explicitly, but we should always pass in the user (from the controller's `RequestWithAuthenticatedUser`) _and_ do all the usual permissions checks - you can mostly copy over the code you've already created in `GeoFeaturesService.updateFeatureForProject()`, for example.